CMakeLists.txt

```diff
@@ -48,19 +48,132 @@ if(POLICY CMP0135)
   cmake_policy(SET CMP0135 NEW)
 endif()
 
+if(WIN32)
```

Comment:
```suggestion
if(WIN32 AND (NOT EXISTS "${CMAKE_TOOLCHAIN_FILE}" OR NOT "${CMAKE_TOOLCHAIN_FILE}" MATCHES ".*${BUILDENV_NAME}.*"))
```

Reply:
`${CMAKE_TOOLCHAIN_FILE}` will never match `${BUILDENV_BASEPATH}/${BUILDENV_NAME}`, as the one is a file and the other a directory.
In my case `D:\mixxx\buildenv\mixxx-deps-2.5-x64-windows-c15790e\scripts\buildsystems\vcpkg.cmake` vs `D:\mixxx\buildenv\mixxx-deps-2.5-x64-windows-c15790e\`

What do you want to achieve with this condition?